### PR TITLE
Fix tests by using API Key for CI

### PR DESCRIPTION
**Pull request by TylerLeonhardt**
## PR Summary

Fixes the tests by using an API Key for CI so that we aren't rate limited by GitHub's APIs.

## PR Checklist

Note: Tick the boxes below that apply to this pull request by putting an `x` between the square brackets.
Please mark anything not applicable to this PR `NA`.

- [x] PR has a meaningful title
- [x] Summarized changes
- [x] PR has tests
- [x] This PR is ready to merge and is not work in progress
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready

### CODE REVIEW

File: src/features/UpdatePowerShell.ts [hunk 1]
```diff
@@ -16,15 +16,27 @@ const PowerShellGitHubPrereleasesUrl =
...
+            };
+        }
+
         // Fetch the latest PowerShell releases from GitHub.
         let releaseJson: any;
         if (preview) {
             // This gets all releases and the first one is the latest prerelease if
             // there is a prerelease version.
-            releaseJson = (await fetch(PowerShellGitHubPrereleasesUrl)
+            releaseJson = (await fetch(PowerShellGitHubPrereleasesUrl, requestConfig)
```
Comment (rjmholt) [inline src/features/UpdatePowerShell.ts]: Isn't the point of `await` to obviate the need for `then()`?

Comment (TylerLeonhardt) [reply src/features/UpdatePowerShell.ts]: It's either this or double awaiting which I felt was a tad less clear with the extra parens and less verbose than an intermediate variable. I can change it if you'd like.

```js
await (await fetch(PowerShellGitHubReleasesUrl, requestConfig)).json()

// or

const res = await fetch(PowerShellGitHubReleasesUrl, requestConfig);
releaseJson = await res.json();
```

Comment (rjmholt) [reply src/features/UpdatePowerShell.ts]: No no, no need to derail the PR

**Comment by TylerLeonhardt** [comment]
@rjmholt refactored for better error handling. I ran CI like 5-8 times and green every time.

**Pull request merged**